src/execution/vm/bytecode_generator.cpp

```diff
@@ -1887,6 +1887,100 @@ void BytecodeGenerator::VisitBuiltinArithmeticCall(ast::CallExpr *call, ast::Bui
   GetExecutionResult()->SetDestination(dest);
 }
 
+void BytecodeGenerator::VisitBuiltinAtomicAndCall(ast::CallExpr *call) {
+  auto args = call->Arguments();
+  LocalVar dest = VisitExpressionForRValue(args[0]);
+  LocalVar val = VisitExpressionForRValue(args[1]);
+  LocalVar ret;
+  auto return_used = GetExecutionResult() != nullptr;
+  if (return_used) {
+    ret = GetExecutionResult()->GetOrCreateDestination(call->GetType());
+    GetExecutionResult()->SetDestination(ret);
+  } else {
+    ret = GetCurrentFunction()->NewLocal(call->GetType());
+  }
+
+  switch (args[1]->GetType()->GetSize()) {
+    case 1:
+      GetEmitter()->Emit(Bytecode::AtomicAnd1, dest, val, ret);
+      break;
+    case 2:
+      GetEmitter()->Emit(Bytecode::AtomicAnd2, dest, val, ret);
+      break;
+    case 4:
+      GetEmitter()->Emit(Bytecode::AtomicAnd4, dest, val, ret);
+      break;
+    case 8:
+      GetEmitter()->Emit(Bytecode::AtomicAnd8, dest, val, ret);
+      break;
+    default:
+      UNREACHABLE("Unexpected operand size for atomicAnd!");
+  }
+}
+
+void BytecodeGenerator::VisitBuiltinAtomicOrCall(ast::CallExpr *call) {
```

Comment:
I think you can combine this with the previous `BuiltinAtomicAndCall()`.

Reply:
I couldn't think of an easy way to combine this with the other call without essentially creating a switch/if statement inside of the combined function that personally I think would be harder to read because it leads to a nested if/switch (the bytecodes are specialized to both operator and operand size).

Reply:
Since the only difference is the bytecode you're using, can you combine them and choose between the two codes?
```
if (size == N) {
  auto bytecode = builtin == Builtin::AtomicOr ? Bytecode::AtomicOr[N] : Bytecode::AtomicAnd[N];
  GetEmitter()->Emit(bytecode, ....);
}
```

A good hint is if you're type-checking shares code, then bytecode generation can probably share code.

Reply:
My only hesitation here is that I'm assuming at some point we'll add `add` and `subtract` (and possibly xor?) at which case that approach gets awkward.  I'm fine doing what you suggested, or I can also bite the bullet and add those atomics as well (should go faster now).